patina_dxe_core/src/allocator/fixed_size_block_allocator.rs

```diff
@@ -767,9 +779,10 @@ unsafe impl Allocator for SpinLockedFixedSizeBlockAllocator {
                 const _: () = assert!(ALIGNMENT.is_multiple_of(align_of::<AllocatorListNode>()));
                 const _: () = assert!(ALIGNMENT.is_multiple_of(UEFI_PAGE_SIZE) && ALIGNMENT > 0);
 
-                // As a matter of policy, allocate at least `MIN_EXPANSION` memory and ensure the size is
-                // aligned to `ALIGNMENT`.
-                let mut allocation_size = max(additional_mem_required, MIN_EXPANSION);
+                // As a matter of policy, allocate at least the minimum expansion amount of memory and ensure the
+                // size is aligned to `ALIGNMENT`.
+                let mut allocation_size =
+                    max(additional_mem_required, min_expansion_for_memory_type(self.lock().memory_type()));
```

Comment:
this will require taking the lock to get the memory type, which slightly (maybe imperceptibly) increases the runtime cost. It's in the expansion flow which is not fast path already; but noting that this isn't zero-cost. If you use a const generic to spec the MIN_EXPANSION size, you could avoid this.

Reply:
I'll update it to use a const generic on Monday.